api/vulnerability_exceptions.go

```diff
@@ -0,0 +1,453 @@
+//
+// Author:: Darren Murray (<darren.murray@lacework.net>)
+// Copyright:: Copyright 2021, Lacework Inc.
+// License:: Apache License, Version 2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package api
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+// VulnerabilityExceptionsService is the service that interacts with
+// the VulnerabilityExceptions schema from the Lacework APIv2 Server
+type VulnerabilityExceptionsService struct {
+	client *Client
+}
+
+// vulnerabilityExceptionResourceScope is an interface for the 2 types of vulnerability exceptions resource scopes:
+// 'VulnerabilityExceptionContainerResourceScope' or 'VulnerabilityExceptionHostResourceScope'
+type vulnerabilityExceptionResourceScope interface {
+	Type() vulnerabilityExceptionType
+	Scope() VulnerabilityExceptionResourceScope
+}
+
+// vulnerabilityExceptionReason represents the types of vulnerability exceptions reasons:
+//'False Positive', 'Accepted Risk', 'Compensating Controls', 'Fix Pending' or 'Other'
+type vulnerabilityExceptionReason int
+
+const (
+	VulnerabilityExceptionReasonAcceptedRisk vulnerabilityExceptionReason = iota
+	VulnerabilityExceptionReasonAcceptedFalsePositive
+	VulnerabilityExceptionReasonCompensatingControls
+	VulnerabilityExceptionReasonFixPending
+	VulnerabilityExceptionReasonOther
+	VulnerabilityExceptionReasonUnknown
+)
+
+var VulnerabilityExceptionReasons = map[vulnerabilityExceptionReason]string{
+	VulnerabilityExceptionReasonAcceptedRisk:          "Accepted Risk",
+	VulnerabilityExceptionReasonAcceptedFalsePositive: "False Positive",
+	VulnerabilityExceptionReasonCompensatingControls:  "Compensating Controls",
+	VulnerabilityExceptionReasonFixPending:            "Fix Pending",
+	VulnerabilityExceptionReasonOther:                 "Other",
+	VulnerabilityExceptionReasonUnknown:               "",
```

Comment:
Should we send something to the APIs so that they tell us that we need to set it, I think if we send an empty string it will just say that an exception name is required. 🤔
```suggestion
	VulnerabilityExceptionReasonUnknown:               "Unknown",
```

Reply:
👍